### ISSUE TITLE
MAP is nan in measure_map.py

### ISSUE THREAD

**Issue by Tangzy7** [yhenon/keras-frcnn#41]
The ap is NaN because in a specific class, T is a zero array like [0,0] and P is [0.7,0.9].  That mean the FN+TP = 0 and F And I think the code does not deal with this situation.

For this situation, the precision-recall curve is the x axis. So the AP(area under curve) is either 0 or inf？

**Comment by yhenon** [comment]
That's correct. Any suggestions on how to handle this? I don't think the AP can be measured if there are no ground truth boxes for that class.

**Comment by chairath** [comment]
@yhenon I checked the code of measure_map.py, i dont understand what  gt_box['difficult'] came from ? Can you explain this one a little bit, I got the error from here.

Thank a lots

**Comment by yhenon** [comment]
pascal VOC dataset has `difficult` annotations  that affect evaluation. Basically these indicate boxes that are not needed to be detected - you are not penalised for missing them (but detection them counts as a true positive).
